In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression

df = pd.read_csv("air_fryers_clean_brand_year.csv")

df["log_share"] = np.log(df["brand_share"])

df.head()

,category,year,brand,purchase_count,product_count,avg_price,avg_rating,compact_share,dual_basket_share,oven_style_share,rotisserie_share,window_share,market_purchases,brand_share,log_brand_share,log_share
0,air_fryers,2019,chefman,1146,10,72.963695,4.434119,1.000000,0.0,0.780977,0.243455,0.184119,15076,0.076015,-2.576826,-2.576826
1,air_fryers,2019,cosori,11,2,159.990000,4.581818,1.000000,0.0,0.090909,0.090909,0.000000,15076,0.000730,-7.222964,-7.222964
2,air_fryers,2019,cuisinart,1616,22,229.465274,4.481312,0.993812,0.0,0.889851,0.000000,0.000000,15076,0.107190,-2.233150,-2.233150
3,air_fryers,2019,dash,3011,19,55.176333,4.390767,1.000000,0.0,0.973431,0.000000,0.000000,15076,0.199721,-1.610832,-1.610832
4,air_fryers,2019,gowise usa,4405,45,83.575551,4.552259,0.999773,0.0,0.129398,0.128490,0.000000,15076,0.292186,-1.230364,-1.230364


In [18]:
exclude_cols = [
    "brand",
    "year",
    "brand_share",
    "log_brand_share",
    "avg_price",
    "avg_rating",
    "log_share"
]

feature_cols = [col for col in df.columns if col not in exclude_cols]

feature_cols = df[feature_cols].select_dtypes(include="number").columns.tolist()

In [19]:
brand_dummies = pd.get_dummies(df["brand"], drop_first=True)
year_dummies = pd.get_dummies(df["year"], drop_first=True)

brand_dummies.head()
year_dummies.head()

,2020,2021,2022,2023
0,False,False,False,False
1,False,False,False,False
2,False,False,False,False
3,False,False,False,False
4,False,False,False,False


In [20]:
X = pd.concat([
    df[["avg_price", "avg_rating"]],
    df[feature_cols],
    brand_dummies,
    year_dummies
], axis=1)

X.columns = X.columns.astype(str)

y = df["log_share"]

X.head()

,avg_price,avg_rating,purchase_count,product_count,compact_share,dual_basket_share,oven_style_share,rotisserie_share,window_share,market_purchases,...,gowise usa,instant_pot,ninja,nuwave,oster,ultrean,2020,2021,2022,2023
0,72.963695,4.434119,1146,10,1.000000,0.0,0.780977,0.243455,0.184119,15076,...,False,False,False,False,False,False,False,False,False,False
1,159.990000,4.581818,11,2,1.000000,0.0,0.090909,0.090909,0.000000,15076,...,False,False,False,False,False,False,False,False,False,False
2,229.465274,4.481312,1616,22,0.993812,0.0,0.889851,0.000000,0.000000,15076,...,False,False,False,False,False,False,False,False,False,False
3,55.176333,4.390767,3011,19,1.000000,0.0,0.973431,0.000000,0.000000,15076,...,False,False,False,False,False,False,False,False,False,False
4,83.575551,4.552259,4405,45,0.999773,0.0,0.129398,0.128490,0.000000,15076,...,True,False,False,False,False,False,False,False,False,False


In [21]:
model = LinearRegression()
model.fit(X, y)

results = pd.DataFrame({
    "variable": X.columns,
    "coefficient": model.coef_
})

results

,variable,coefficient
0,avg_price,-0.038265
1,avg_rating,1.633819
2,purchase_count,0.000360
3,product_count,0.016530
4,compact_share,3.356568
5,dual_basket_share,5.471489
6,oven_style_share,-0.022199
7,rotisserie_share,0.506475
8,window_share,3.277291
9,market_purchases,-0.000030


#### Answers
1. The estimated price coefficient is approximately −0.038.
2. Yes, it is negative, which is important because it matches normal demand behavior: when price goes up, demand should go down.